sirepo/template/code_variable.py

```diff
@@ -63,6 +66,29 @@ def eval_var_with_assert(self, expr):
         assert not err, err
         return float(v)
 
+    def get_application_data(self, args, schema, ignore_array_values=False):
+        from sirepo import simulation_db
+        if args.method == 'rpn_value':
+            if ignore_array_values and re.search(r'^\{.*\}$', args.value):
+                # accept array of values enclosed in curly braces
+                args.result = ''
+                return True
+            v, err = self.eval_var(args.value)
+            if err:
+                args.error = err
+            else:
+                args.result = v
+            return True
+        if args.method == 'recompute_rpn_cache_values':
+            self.recompute_cache(args.cache)
+            return True
+        if args.method == 'validate_rpn_delete':
+            model_data = simulation_db.read_json(
+                simulation_db.sim_data_file(args.simulationType, args.simulationId))
+            args.error = self.validate_var_delete(args.name, model_data, schema)
+            return True
+        return False
```

Comment:
Shouldn't this throw an exception? see false case below which is not handled properly.

Reply:
The True/False is indicating if the data.method was handled by the call. If it threw an exception, it would always have to be the very last call in get_application_data().

Reply:
Or, https://github.com/radiasoft/sirepo/pull/2721#discussion_r463821235 would dispatch based on method. Each method would map to a function on a class (ApplicationData). code_variables.get_application_data would be a base class.